pkgs/tools/security/kdigger/default.nix

```diff
@@ -0,0 +1,76 @@
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles, fetchpatch }:
+
+buildGoModule rec {
+  pname = "kdigger";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "quarkslab";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-j4HIwfRIUpV25DmbQ+9go8aJMEYaFDPxrdr/zGWBeVU=";
+    # populate values that require us to use git. By doing this in postFetch we
+    # can delete .git afterwards and maintain better reproducibility of the src.
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git rev-parse HEAD > $out/COMMIT
+      find "$out" -name .git -print0 | xargs -0 rm -rf
+    '';
+  };
+  vendorSha256 = "sha256-3vn3MsE/4lBw89wgYgzm0RuJJ5RQTkgS6O74PpfFcUk=";
+
+  patches = [
+    (fetchpatch {
+      name = "simplify-ldflags.patch";
+      url = "https://github.com/quarkslab/kdigger/pull/2.patch";
+      sha256 = "sha256-d/NdoAdnheVgdqr2EF2rNn3gJvbjRZtOKFw2DqWR8TY=";
+    })
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  CGO_ENABLED = 0;
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/quarkslab/kdigger/commands.VERSION=v${version}"
+  ];
+
+  preBuild = ''
+    ldflags+=" -X github.com/quarkslab/kdigger/commands.GITCOMMIT=$(cat COMMIT)"
+    ldflags+=" -X github.com/quarkslab/kdigger/commands.BUILDERARCH=$(uname -m)"
```

Comment:
```suggestion
```
Thats impure.

Reply:
How so, shouldn't it always be x86-64 on an x86_64 builder and aarch64 if crosscompiled for x86_64-linux from an aarch64-linux platform?

It doesnt have other details like kernel label/name/whatever, date, etc

Reply:
I guess I can pull an equivalent value from stdenv somewhere

Reply:
Yeah, you're right. I thought it includes the x86_64 variants but we still shouldn't use it because it depends on the build machine and not the nixpkgs platform. Also it probably breaks easily on binfmt.